README.md

```diff
@@ -18,3 +18,45 @@ Resume a training run specified by `<config_file>` starting from `<run_id>/<chec
 ```
 python src/ --config conf/train/<config_file>.yml --resume models/<run_id>/<checkpoint_file>
 ```
+
+
+## Extending this repo
+
+We follow the following flow for objects in this repository:
+![image of the flow map](<readme_assets/flow map.png>)
+
+These initialized classes are passed the dictionary parsed directly from yaml as arguments. We recommend reading examples for your respective function class in [conf/](https://github.com/in-context-learning-2024/in-context/blob/main/conf/)
+
+Make sure to initialize your parent class with `super().__init__(...)` in both cases
+
+We try to make it easy to contribute by [adding models](#adding-models) or [function classes](#adding-function-classes):
+
+
+---
+### Adding Models
```

Comment:
should we add an explanation of how to get yaml files to 'see' their added context model? like, adding it to the \_\_init\_\_.py file and then adding it to models.yml?

Reply:
this is a good point. added